pkg/subscription/init.go

```diff
@@ -0,0 +1,33 @@
+package subscription
+
+// InitPayload is a structure that is parsed from the websocket init message payload.
+type InitPayload map[string]interface{}
```

Comment:
This assumes that InitPayload is a map. What if it's any other valid JSON? Could it be an Array or a string, bool or number?

I see no other issues with this PR, but would suggest that you rewrite this to accept any kind of JSON, or maybe even a binary payload. What are your thoughts on this? (just making it a bit more generic) Nothing against the Authorization helper method. It can live in some way, but the init handler should be unopinionated.

Reply:
Looking at the type of connectionParams in `subscription-transport-ws` it seems to always be a map:
```
export declare type ConnectionParams = {
    [paramName: string]: any;
};
```

Reply:
Ok, coming back to this one finally. I changed InitPayload to be a json.RawMessage and then trying to unmarshal it when getting a string value. That should make it handle any kind of value, right?